portable/NetworkInterface/TM4C/NetworkInterface.c

```diff
@@ -0,0 +1,599 @@
+                    EMAC_CONFIG_7BYTE_PREAMBLE |
+                    EMAC_CONFIG_IF_GAP_96BITS |
+                    EMAC_CONFIG_USE_MACADDR0 |
+                    EMAC_CONFIG_SA_FROM_DESCRIPTOR |
+                    EMAC_CONFIG_BO_LIMIT_1024
+                ),
+                (
+                    EMAC_MODE_RX_STORE_FORWARD |
+                    EMAC_MODE_TX_STORE_FORWARD |
+                    EMAC_MODE_RX_THRESHOLD_64_BYTES |
+                    EMAC_MODE_TX_THRESHOLD_64_BYTES),
+                0);
+
+
+        // Clear any stray PHY interrupts that may be set.
+        ui16Val = MAP_EMACPHYRead(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_MISR1);
+        ui16Val = MAP_EMACPHYRead(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_MISR2);
+
+        // Configure and enable the link status change interrupt in the PHY.
+        ui16Val = MAP_EMACPHYRead(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_SCR);
+        ui16Val |= (EPHY_SCR_INTEN_EXT | EPHY_SCR_INTOE_EXT);
+        MAP_EMACPHYWrite(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_SCR, ui16Val);
+        MAP_EMACPHYWrite(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_MISR1, (EPHY_MISR1_LINKSTATEN |
+                     EPHY_MISR1_SPEEDEN | EPHY_MISR1_DUPLEXMEN | EPHY_MISR1_ANCEN));
+
+        // Read the PHY interrupt status to clear any stray events.
+        ui16Val = MAP_EMACPHYRead(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_MISR1);
+
+        // Set MAC filtering options.  We receive all broadcast and mui32ticast
+        // packets along with those addressed specifically for us.
+        MAP_EMACFrameFilterSet(EMAC0_BASE, (EMAC_FRMFILTER_HASH_AND_PERFECT |
+                           EMAC_FRMFILTER_PASS_MULTICAST));
+
+        // Set the MAC address
+        MAP_EMACAddrSet(EMAC0_BASE, 0, &mac_address_bytes[0]);
+
+        // Clears any previously asserted interrupts
+        MAP_EMACIntClear(EMAC0_BASE, EMACIntStatus(EMAC0_BASE, false));
+
+        // Initialize the DMA descriptors
+        _dma_descriptors_init();
+
+        // Enable TX/RX
+        MAP_EMACTxEnable(EMAC0_BASE);
+        MAP_EMACRxEnable(EMAC0_BASE);
+
+        // Set the interrupt to a lower priority than the OS scheduler interrupts
+        MAP_IntPrioritySet(INT_EMAC0,  (6 << (8 - configPRIO_BITS)));
+
+        // Enable the Ethernet RX and TX interrupt source.
+        MAP_EMACIntEnable(EMAC0_BASE, (EMAC_INT_RECEIVE | EMAC_INT_TRANSMIT |
+                      EMAC_INT_TX_STOPPED | EMAC_INT_RX_NO_BUFFER |
+                      EMAC_INT_RX_STOPPED | EMAC_INT_PHY));
+
+        // Enable EMAC interrupts
+        MAP_IntEnable(INT_EMAC0);
+
+        // Enable all processor interrupts.
+        MAP_IntMasterEnable();
+
+        // Tell the PHY to start an auto-negotiation cycle.
+        MAP_EMACPHYWrite(EMAC0_BASE, PHY_PHYS_ADDR, EPHY_BMCR, (EPHY_BMCR_ANEN |
+                EPHY_BMCR_RESTARTAN));
+    }
+
+    return success;
+}
+
+BaseType_t xNetworkInterfaceOutput(NetworkBufferDescriptor_t * const pxNetworkBuffer, BaseType_t xReleaseAfterSend)
+{
```

Comment:
The parameter `xReleaseAfterSend` will be non-zero in case the ownership of the network packet is passed to the driver.
This is a zero-copy driver, and so `xReleaseAfterSend` sill always be true. So it is OK not to check it. OR at most you could:
~~~c
    /* Check if `ipconfigZERO_COPY_TX_DRIVER` was set in FreeRTOSIPConfig.h. */
    configASSERT( xReleaseAfterSend != 0 );
~~~

Reply:
Got it. I'll add the assert.